src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -242,7 +236,12 @@ public override BoundNode VisitLambda(BoundLambda node)
             try
             {
                 _factory.CurrentFunction = node.Symbol;
-                return base.VisitLambda(node);
+                var visited = (BoundLambda)base.VisitLambda(node);
+                if (RewriteNullChecking(visited.Body) is BoundBlock newBody && !(newBody is null))
```

Comment:
`is BoundBlock` already ensures the expression is non-null, so no need for `!(newBody is null)`